drivers/drivers.go

```diff
@@ -0,0 +1,11 @@
+package drivers
+
+// The Driver interface is the set of functions required to start, manage, and verify a Kubernetes cluster. Inspired by the steps used in kube-up.sh, with extensions for management and scaling of clusters.
+type Driver interface {
+	GenerateCerts()
+	GetTokens()
+	ProvisionMaster()
+	ConfigureMaster()
+	ProvisionNode()
```

Comment:
that's reasonable. feel free to tweak and add as you see fit. For now, getting in the main steps for creation for a driver.